pkg/prometheus/promcfg.go

```diff
@@ -4505,6 +4507,10 @@ func (cg *ConfigGenerator) appendOTLPConfig(cfg yaml.MapSlice) (yaml.MapSlice, e
 		return cfg, fmt.Errorf("OTLP configuration is only supported from Prometheus version 2.55.0")
 	}
 
+	if otlpConfig.TranslationStrategy != nil && nameValidationScheme != nil && *nameValidationScheme == monitoringv1.LegacyNameValidationScheme {
```

Comment:
we need to check both values
```suggestion
	if ptr.Deref(otlpConfig.TranslationStrategy, "") == monitoringv1.NoUTF8EscapingWithSuffixes && ptr.Deref(nameValidationScheme, "") == monitoringv1.LegacyNameValidationScheme {
```